pkg/proxy/policy/policy.go

```diff
@@ -24,51 +24,57 @@ func RegisterRule(rulename string, constructor func() Rule) {
 // Policy contains a list of Rules that will be applied to the request.
 type Policy struct {
 	// AnyOf expects incoming requests to satisfy one of these Rules.
-	AnyOf []Rule
+	AnyOf []Rule `json:"anyOf"`
 }
 
 // UnmarshalJSON implements the json.Unmarshaler interface for the Policy class.
 // Expects rule_type to specify an existing rule in the rule registry.
 func (p *Policy) UnmarshalJSON(data []byte) error {
 	var policywrapper struct {
 		Policy struct {
-			AnyOf []json.RawMessage `json:"AnyOf"`
-		} `json:"Policy"`
+			AnyOf []json.RawMessage
+		}
 	}
 	if err := json.Unmarshal(data, &policywrapper); err != nil {
 		return err
 	}
+	rules, err := unmarshalRuleType(policywrapper.Policy.AnyOf)
```

Comment:
can you make this function scalar? that way you can just progressively build AnyOf instead of assigning to it at the end

Reply:
wait I don't think this got added. the loadRules still operates on the array, right? I think it should just be loadRule and take the RawMessage and return the Rule.